hubspot/client_test.go

```diff
@@ -54,6 +55,49 @@ var _ = Describe("Client", func() {
 
 	})
 
+	Describe("Email", func() {
+		var (
+			server *httptest.Server
+		)
+
+		BeforeEach(func() {
+			server = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				if strings.Contains(r.RequestURI, "/email/public/v1/singleEmail/send") {
+					w.WriteHeader(http.StatusOK)
+					w.Write([]byte(singleEmailResp))
+					requestBodyMatchesJsonTemplate(r.Body, emailRequest)
```

Comment:
I think you can remove your func `requestBodyMatchesJsonTemplate` by just reading the body and use MatchJSON
```suggestion
					b, err := ioutil.ReadAll(r.Body)
					Expect(err).NotTo(HaveOccurred())
					Expect(b).To(MatchJSON(emailRequest))
```

Reply:
ohhh did know there is MatchJSON TIL!